### ISSUE TITLE
If you create Uturn with TMPE lane connects, U-turn rule will remain after you remove that lane connection

### ISSUE THREAD

**Issue by kianzarrin** [CitiesSkylinesMods/TMPE#901]
problems:
- U-turn most not be configurable if all lanes have connections
- U-turn lane connection must only influence the `DefaultUturnAllowd` and should not call `SetUturnAllowed`

I will complete this when I have time.

**Comment by kianzarrin** [comment]
@aubergine10 do you think removing lane connection should reset both lane arrows and u-turn to default (u-turn is kind of like a lane arrow anyway) ?

**Comment by aubergine10** [comment]
Yes.

I always found it really icky that lane connectors can be used to create u-turn, given that we have dedicated u-turn feature in junction restrictions. Using lane connector in that way can easily result in a lane that can _only_ be used for u-turns (if user doesn't connect to anything else), which is not somthing that exists IRL as far as I know?

Also, do we have a special u-turn lane arrow (if so I never noticed it)? 

IMO the junction restriction u-turn feature should be single source of truth for u-turns. It could be that icon appears when using lane connectors / arrows tools also?

**Comment by kianzarrin** [comment]
@aubergine10 
> I always found it really icky that lane connectors can be used to create u-turn, given that we have dedicated u-turn feature in junction restrictions.

lane connector can be used to force cars to turn yet we have a dedicated lane arrow tool. I found u-turn lane connector useful to have control over target lane of the u-turn. 

more importantly how can I tell cars to turn left and + u-turn if there lane connectors did not handle u-turn? at best it will be confusing.

> Also, do we have a special u-turn lane arrow button (if so I never noticed it)?

No. I edited and clarified the mod description


> IMO the junction restriction u-turn feature should be single source of truth for u-turns. It could be that icon appears when using lane connectors / arrows tools also?

I think there should be an overlay in lane connector tool so that ppl would at least know when u-turn as been disabled. it would be also could to move the u-turn feature to the lane arrow tool. but it should be for the whole segment. currently everyone can do u-turn only from the inner must lane (unless lane connectors are used) except for SOS.

**Comment by kianzarrin** [comment]
@aubergine10 should removing lane connection reset lane arrows back to default?

**Comment by aubergine10** [comment]
> @aubergine10 should removing lane connection reset lane arrows back to default?

Yes, I think so. When I remove a lane connection, it's because I want to get back to vanilla routing.

> I think there should be an overlay in lane connector tool so that ppl would at least know when u-turn as been disabled. it would be also could to move the u-turn feature to the lane arrow tool.

This could be problematic, becuse TM:PE features can be enabled/disabled in mod settings:

![image](https://user-images.githubusercontent.com/1386719/145668656-fc6fc6b6-b82a-4d39-8af4-f290c02f7921.png)

What if user has disabled junction restrictions? In that situation, we could show the U-Turn icon, and have it toggle a u-turn lane connection maybe?

Also, with lane arrows, this seems odd - there is no right-turn, but I can enable right turn lane arrow:

![image](https://user-images.githubusercontent.com/1386719/145668774-c435aecd-d9c4-458a-9df7-edad3d4eb595.png)

Does that mean traffic can u-turn? This is why I always ask for 'grey' lane connectors to appear when using lane arrow tool, ie. visualise what the chosen arrows are actually doing.

Also, if a u-turn is defined on the lane arrow screen, how is it implemented if the lane connectors & junction restriction tools are disabled? Would we now have 3 different ways to control u-turns?

**Comment by kianzarrin** [comment]
@aubergine10 
> Also, if a u-turn is defined on the lane arrow screen, how is it implemented if the lane connectors & junction restriction tools are disabled? Would we now have 3 different ways to control u-turns?

maybe U-turn should not be managed by junction restrictions anymore. it just that u-turn is very close to lane arrow so it makes sense to put it there. Not sure if its possible to make it work without making it confusing or backward-incompatible.

also u-turn could show only for the inner most lane rather than the whole segment. but that is just UI interface.

> Also, with lane arrows, this seems odd - there is no right-turn, but I can enable right turn lane arrow:
> Does that mean traffic can u-turn? 

I recently fixed an issue regarding this. you don't need the right turn arrow for u-turn anymore. But if we show u-turn rather than right-turn in the UI that would make more sense. Adaptive-Networks mod can take care of the props.
